src/utils/reeds_data_parsing.jl

```
@@ -363,6 +445,218 @@ function process_vg(
+    # and seasonal capacity adjustment factor on the nameplate capacity for dispatch limits. 
+    # For each month, (i) energy budget is calculated based on number of hours in the month, and the 
+    # season the month belongs to; and (ii) dispatch limit is calculated based on the season of the month.
+    for (idx, row) in enumerate(DataFrames.eachrow(disp_combined_caps))
+        reg_plant_subset =
+            filter(x -> (x.r == row.r && x.i == row.i), hydcf)[:, [:month, :value]]
+        monthly_energy = zeros(timesteps_year)
+        dispatch_limit = zeros(timesteps_year)
+        energy_cap = zeros(timesteps_year)
+        for (idx_mon, mon_row) in enumerate(DataFrames.eachrow(monthhours))
+            #@debug reg_plant_subset[findall(x->startswith(mon_row.season,x),reg_plant_subset.szn),:value]
+            try
+                reqd_slice = (mon_row.cumhrs - mon_row.numhrs + 1):(mon_row.cumhrs)
+                monthly_energy[first(reqd_slice)] = (mon_row.numhrs * filter(
+                    x -> (x.month == mon_row.month),
+                    reg_plant_subset,
+                )[
+                    :,
+                    :value,
+                ] * row.MW_sum)[1]
+
+                energy_cap[reqd_slice] .= (mon_row.numhrs * filter(
+                    x -> (x.month == mon_row.month),
+                    reg_plant_subset,
+                )[
+                    :,
+                    :value,
+                ] * row.MW_sum)[1]
+
+                capacity_adjust = hydcapadj[
+                    (hydcapadj.i .== row.i) .&& (hydcapadj.r .== row.r),
+                    [:month, :value],
+                ]
+                dispatch_limit[reqd_slice] .=
+                    (filter(x -> (x.month == mon_row.month), capacity_adjust)[
+                        :,
+                        :value,
+                    ] * row.MW_sum)[1]
+            catch e
+                @error "$(row.r),$(row.i),$(e)"
```

Should we get rid of this try/catch and let it fail if there's an error? I doubt people will check every log file carefully enough to notice it.

---

I second this.

---

Doing this only if we catch a `BoundsError` now. Added a TODO item so its taken care of in ReEDS.